src/commands/createFunction/createFunction.ts

```diff
@@ -54,3 +43,42 @@ export async function createFunction(
     await wizard.prompt(actionContext);
     await wizard.execute(actionContext);
 }
+
+async function getWorkspacePath(actionContext: IActionContext): Promise<string> {
+    let folder: WorkspaceFolder | undefined;
+    if (!workspace.workspaceFolders || workspace.workspaceFolders.length === 0) {
+        const message: string = localize('noWorkspaceWarning', 'You must have a project open to create a function.');
+        const newProject: MessageItem = { title: localize('createNewProject', 'Create new project') };
+        const openExistingProject: MessageItem = { title: localize('openExistingProject', 'Open existing project') };
+        const result: MessageItem = await ext.ui.showWarningMessage(message, { modal: true }, newProject, openExistingProject);
+
+        if (result === newProject) {
+            // don't wait
+            commands.executeCommand('azureFunctions.createNewProject');
```

Comment:
I'm wondering why you're using `executeCommand`  instead of calling the function (`createNewProject`) directly.

Reply:
Mostly I want a separate telemetry event instead of including it with the createFunction event